index.js

```diff
@@ -889,10 +889,8 @@ let createContainer = ({resolvers = [], dependencies = {}, factories, definition
 
         if (!definition.reuse) {
             destroyObject(instance, options);
-            definition.instance = null;
-        } else {
-            delete definition.instance;
         }
+        definition.instance = null;
```

Comment:
For example we have `B: '!A'` and `C: '!A'`, then:

Session 1:

```
di('B');
```

We create instance with `B` dependencies and put it on `B` and `A`.

Session 2:


```
di('C');
```

`C` has no instance and see through `__proto__` instance of `A`, it detect, that it was created in another session and make `updateDependences` on it. So `A` and `C` now has dependencies of `C` and it marked as updated. `B` has same instance too, but it will be destroyed at the end of process. We delete instance property from `B` definition.  


Session 3:

```
di('B');
```

Because `instance` of `B` definition is not set, it found `instance` in `A` by `__proto__` link. All process will be same as in session 2.

Situation, when we request `B` and `C` dependency in same session is incorrect, because we have undefined behavior of which dependencies to use (instance shared between all of them). Mister @next0 suggest to check this and throw Error, but it not implemented yet.